test/src/test/java/org/corfudb/integration/LogReplicationSinkWriterIT.java

```diff
@@ -0,0 +1,289 @@
+     *    2). Table007 and Table008 - open on source side only
+     * 5. Verify the table records and contents after log entry sync
+     */
+    @Test
+    @SuppressWarnings("checkstyle:magicnumber")
+    public void testFilterAndApplyRegistryTableEntries() throws Exception {
+        setupActiveAndStandbyCorfu();
+        openLogReplicationStatusTable();
+
+        // Initial setup for streams to replicate for both source and sink side
+        int numTableActive = 8;
+        int numTableStandby = 8;
+        mapNameToMapActive = new HashMap<>();
+        mapNameToMapStandby = new HashMap<>();
+        setupStreamsToReplicate(numTableActive, true);
+        setupStreamsToReplicate(numTableStandby, false);
+
+        /* The first 4 tables are for snapshot sync */
+
+        // Table001 and Table002 - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
+        // Table003 and Table004 - Open only in Source side. Verify that their records are replicated to Sink.
+        prepareTablesToVerify(1, 2, 4);
+
+        // Add Data for Snapshot Sync
+        writeToActive(0, NUM_WRITES);
+        // Confirm data does exist on Active Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapActive.values()) {
+            assertThat(map.count()).isEqualTo(NUM_WRITES);
+        }
+
+        // Confirm data does not exist on Standby Cluster
+        for(Table<StringKey, IntValueTag, Metadata> map : mapNameToMapStandby.values()) {
+            assertThat(map.count()).isEqualTo(0);
+        }
+
+        // Set the plugin config file path for Source before starting the process, such that Source side
+        // log replicator could read the preset streams to replicate from STREAMS_TEST_TABLE.
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_ACTIVE;
+        startActiveLogReplicator();
+
+        // Switch the plugin file path to Sink side for the same reason.
+        pluginConfigFilePath = TEST_PLUGIN_CONFIG_PATH_STANDBY;
+        startStandbyLogReplicator();
+
+        // Verify snapshot sync completed and cluster went into log entry sync
+        verifyInLogEntrySyncState();
+        // Verify Sink side records for Table001 and Table002 are retained, and records for Table003
+        // and Table004 are replicated.
+        // Also verify all the table's contents for snapshot sync
+        verifyTableContentsAndRecords(1, 2, 4);
+
+
+        /* The last 4 tables are for log entry sync */
+
+        // Stop LR to modify Sink side registry table in advance
+        stopActiveLogReplicator();
+        // Clear these two map as different set of tables will be verified
+        mapNameToMapActive.clear();
+        mapNameToMapStandby.clear();
+
+        // Table005 and Table006 - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
```

Comment:
nit - .. retained on the registry table.  Same comment for the comment above for snapshot sync